### PR TITLE
Fix type inference for set/delete attr.

### PR DESCRIPTION
**Pull request by robertwb**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @yeandy

File: sdks/python/apache_beam/typehints/opcodes.py [hunk 1]
```diff
@@ -241,8 +241,8 @@ def dup_topx(state, arg):
   state.stack += state[-arg:]
 
 
-store_attr = pop_top
-delete_attr = nop
```
Comment (yeandy) [inline sdks/python/apache_beam/typehints/opcodes.py:245]: Can you provide more context on this change? Where there any previous issues/failures?

File: sdks/python/apache_beam/typehints/trivial_inference_test.py [hunk 2]
```diff
@@ -258,6 +258,26 @@ def testGetAttr(self):
...
+      elif flag == 3:
+        glob = 3
+        res = "str"
+      elif flag == 4:
+        del glob
+        res = "another str"
+      return res
+    self.assertReturnType(
+        typehints.Union[int, float, str],
+        fn, [int])
```
Comment (yeandy) [inline sdks/python/apache_beam/typehints/trivial_inference_test.py:279]:
```suggestion

    self.assertReturnType(typehints.Union[int, float, str],fn, [int])
```

File: sdks/python/apache_beam/typehints/trivial_inference_test.py [hunk 3]
```diff
@@ -258,6 +258,26 @@ def testGetAttr(self):
         typehints.Tuple[str, typehints.Any],
         lambda: (typehints.__doc__, typehints.fake))
 
+  def testSetAttr(self):
+    def fn(obj, flag):
+      global glob
```
Comment (yeandy) [inline sdks/python/apache_beam/typehints/trivial_inference_test.py:263]: I see we're getting this lint error: `Global variable 'glob' undefined at the module level (global-variable-undefined)`. Why are we defining a global here? Would defining `glob` outside of `def fn(obj, flag):` fix that? 

I also see there are the `store_global` and `delete_global` opcodes; were you also investigating those?

File: sdks/python/apache_beam/typehints/trivial_inference_test.py (hunk 2 above)
Comment (robertwb) [reply sdks/python/apache_beam/typehints/trivial_inference_test.py]: Done.

File: sdks/python/apache_beam/typehints/trivial_inference_test.py (hunk 3 above)
Comment (robertwb) [reply sdks/python/apache_beam/typehints/trivial_inference_test.py]: Yes, these were correct, but I added a test for them. I've now separated that out into a separate test to be more clear (and disabled the hint warning that isn't useful here).

Comment (yeandy) [reply sdks/python/apache_beam/typehints/trivial_inference_test.py]: Makes sense, thanks!

**Comment by yeandy** [comment]
LGTM

**Review by yeandy — APPROVED** [review]

**Pull request merged**